src/main/kotlin/org/jqassistant/tooling/intellij/plugin/editor/templates/RuleJqaTemplateFileCreation.kt

```diff
@@ -2,166 +2,125 @@ package org.jqassistant.tooling.intellij.plugin.editor.templates
+            Messages.showInfoMessage(project, "$targetName created in ${targetFile.parent}", "File Created")
             VirtualFileManager.getInstance().syncRefresh()
-            openFileAutomatic(project, targetFile)
+            openFile(project, targetFile)
         } else {
-            Messages.showErrorDialog(
-                project,
-                "Template file my_rules.xml not found in plugin resources.",
-                "Error",
-            )
+            // Show an error message if the template resource is not found.
+            Messages.showErrorDialog(project, "Template file not found in resources.", "Error")
         }
     }
 
-    fun isNameValid(name: String) : Boolean{
-        //TODO considering better options for validation
-        //All supported Characters of a file name
-        val validNameRegex = Regex("^[a-zA-Z0-9._-]+$")
-        // Additional check: No control characters or invalid Unicode characters
-        return name.matches(validNameRegex) && name.codePoints().noneMatch { Character.isISOControl(it) }
+    // Opens the newly created file in the editor.
+    private fun openFile(project: Project, file: File) {
+        LocalFileSystem.getInstance().findFileByIoFile(file)?.let {
+            FileEditorManager.getInstance(project).openFile(it, true)
+        } ?: Messages.showErrorDialog(project, "File created but could not be opened.", "Error")
     }
 
-    fun openFileAutomatic(project : Project, newFile : File) {
-        val virtualFile: VirtualFile? = LocalFileSystem.getInstance().findFileByIoFile(File(newFile.path))
-        if (virtualFile != null) {
-            FileEditorManager.getInstance(project).openFile(virtualFile, true)
-        } else {
-            Messages.showErrorDialog(
-                project,
-                "The file was created but could not be opened automatically.",
-                "Error Opening File"
-            )
-        }
+    // Prompts the user for input using a dialog box and returns the input.
+    private fun getUserInput(project: Project, message: String): String? =
+        Messages.showInputDialog(project, message, "Input", Messages.getQuestionIcon())?.trim()
+
+    // Validates the file name to ensure it contains only allowed characters and no control characters.
+    private fun isNameValid(name: String): Boolean {
+        val validChars =
+            (
+                ('a'..'z') + ('A'..'Z') + ('0'..'9') + listOf('.', '_', '-')
+            ).toSet() // Set of allowed characters.
+
+        // Check if all characters are valid and no control characters are present.
+        return name.all { it in validChars } && name.codePoints().noneMatch { Character.isISOControl(it) }
     }
 
-    //Duplication of functions is noted, will be refactored in final product anyway
+    // Determines the appropriate file extension based on the default file name.
+    private fun getFileExtension(): String = if (fileName.endsWith(".yaml")) "" else ".xml"
```

Comment:
Given the user enters `name-of-file.xml`, would this not create a file named `name-of-file.xml.xml`? Is this the intended behavior?

 If no a solution would be to check if the file already ends with `.xml` and return `""` if that is the case